### PR TITLE
Backport of Support x-forwarded-client-cert  into release/1.12.x

### LINKED ISSUE #12852: Support envoy x-forwarded-client-cert header in consul
#### Feature Description

Envoy provides support forwarding and annotating the x-forwarded-client-cert header via the forward_client_cert_details set_current_client_cert_details filter fields. It would be helpful for consul to support this directly in its config. The escape hatches are a bit cumbersome for this purpose. 

#### Use Case(s)
This information is useful for various application side checks and validation. 

#### Implementation proposal

Extend the service registration proxy parameters [structure](https://www.consul.io/docs/connect/registration/service-registration#mesh-gateway-configuration-reference) to include a new structure
```
filter_options = {
     forward_client_cert_details = SANITIZE|FORWARD_ONLY|APPEND_FORWARD|ALWAYS_FORWARD_ONLY
     set_current_client_cert_details = {
         subject = <BOOL>
         cert = <BOOL>
         chain = <BOOL>
         dns = <BOOL>
         uri = <BOOL>
    }
}
```

This is a direct representation of the envoy config fields [extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.ForwardClientCertDetails](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#enum-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-forwardclientcertdetails) and 
[extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.SetCurrentClientCertDetails](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-setcurrentclientcertdetails)

##### Alternatives
We could flatten those fields out and not introduce the filter_options structure, but that seems unnecessarily verbose and clutters the config structure. We may also wish to add other fields at a later date. 

We could also use the ```config``` blob to insert something similar without extending the structure. This structure is described as 'opaque' in the main description but in practice it seems we use that for other proxy specific [overrides](https://www.consul.io/docs/connect/proxies/envoy#escape-hatch-overrides) and would avoid further expansion of the proxy structure. The primary reason to not follow this pattern is that it more complex to do early parameter validation.

### PR DESCRIPTION
**Pull request by hc-github-team-consul-core**
## Backport

This PR is auto-generated from #12878 to be assessed for backporting due to the inclusion of the label backport/1.12.

The below text is copied from the body of the original PR.

---

### Description

Add x-fowarded-client-cert information where appropriate

Envoy provides support forwarding and annotating the x-forwarded-client-cert header via the forward_client_cert_details set_current_client_cert_details filter fields. It would be helpful for consul to support this directly in its config. The escape hatches are a bit cumbersome for this purpose.

### Links
Closes #12852

Asana [consul-x-forwarded-client-cert](https://go.hashi.co/consul-x-fowarded-client-cert)

### PR Checklist

* [x] Unit tests
* [x] End to end tests verifying headers are correct
* [x] external facing docs updated
* [ ] not a security concern
* [ ] checklist [folder](./../docs/config) consulted

### CODE REVIEW

File: website/content/docs/connect/config-entries/mesh.mdx [hunk 1]
```diff
@@ -359,6 +359,21 @@ Note that the Kubernetes example does not include a `partition` field. Configura
...
+      type: 'HTTPConfig: <optional>',
+      description: 'HTTP configuration for the service mesh.',
+      children: [
+        {
+          name: 'SanitizeXForwardedClientCert',
+          yaml: false,
+          type: 'bool: <optional>',
+          description: `Set the envoy \`forward_client_cert_details\` option to \`SANITIZE\` for all proxies. This
+                        configures Envoy to not send the \`x-forwarded-client-cert\` header to the next hop. If
+                        unspecified or \`false\`, the XFCC header is propagated to upstream applications.`,
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/mesh.mdx:373]:
```suggestion
          description: `If configured to `true`, the \`forward_client_cert_details\` option will be to \`SANITIZE\` for all Envoy proxies. As a result, Consul will not include the \`x-forwarded-client-cert\` header in the next hop. 
                        If to \`false\` (default), the XFCC header is propagated to upstream applications.`,
```
Am I understanding this correctly? Is that clearer?

**Review by trujillo-adam — COMMENTED** [review]
Tried to improve the wording.

File: website/content/docs/connect/config-entries/mesh.mdx [hunk 2]
```diff
@@ -368,9 +368,8 @@ Note that the Kubernetes example does not include a `partition` field. Configura
           name: 'SanitizeXForwardedClientCert',
           yaml: false,
           type: 'bool: <optional>',
-          description: `Set the envoy \`forward_client_cert_details\` option to \`SANITIZE\` for all proxies. This
-                        configures Envoy to not send the \`x-forwarded-client-cert\` header to the next hop. If
-                        unspecified or \`false\`, the XFCC header is propagated to upstream applications.`,
+          description: `If configured to `true`, the \`forward_client_cert_details\` option will be to \`SANITIZE\` for all Envoy proxies. As a result, Consul will not include the \`x-forwarded-client-cert\` header in the next hop.
```
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/mesh.mdx:371]:
```suggestion
          description: `If configured to `true`, the \`forward_client_cert_details\` option will be set to \`SANITIZE\` for all Envoy proxies. As a result, Consul will not include the \`x-forwarded-client-cert\` header in the next hop. 
```

**Review by trujillo-adam — APPROVED** [review]
I introduced a typo 😞, but pre-approving

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/662195.

**Pull request merged**